views/forms/BarChartFields.jsx

```diff
@@ -36,6 +42,26 @@ const BarChartFields = ({
           : field
         )
       }))
+  const handleTargetValues = (e, id, tid) => {
+    setFormValues((prevState) => ({
+      ...prevState,
+      fieldLabelValueMap: prevState
+      .fieldLabelValueMap
+        .map((field, idx) => 
+          id === idx ? 
+            { 
+              ...field, 
+              targetValues: field.targetValues
+                .map((target, tidx) => 
+                  tid === tidx 
+                  ? { ...target, value: e.target.value } 
+                  : target
+                ) 
+            }
+          : field
+        )
+    }))
+  }
```

Comment:
This seems like a really complicated change event handler for what I think is updating a single text field at a time? 
What do you think about adding something like [Formik](https://formik.org/) or [ReactHookForm](https://react-hook-form.com/)? I personally have a slight preference for Formik, but both provide a really clean api for handling form state, field value changes and error handling.

Reply:
I agree, it was quite tricky to get this to work smoothly. We have a card to add Form library at some point https://github.com/orgs/AMP-SCZ/projects/2. Right now we're just piloting without validations and form controllers.

Reply:
oh nice! I think it would be worth adding Formik sooner than later. You don't have to set up the error handling/validation right away. Out of the box it would take care of managing form state, so this change handler could at least be simplified and I _think_ might actually not be needed. Happy to pair on setting that up if that would be helpful!